api/agent/drivers/docker/docker.go

```diff
@@ -51,17 +51,79 @@ func (r *runResult) Error() error   { return r.err }
 func (r *runResult) Status() string { return r.status }
 
 type DockerDriver struct {
-	cancel   func()
-	conf     drivers.Config
-	docker   dockerClient // retries on *docker.Client, restricts ad hoc *docker.Client usage / retries
-	hostname string
-	auths    map[string]driverAuthConfig
-	pool     DockerPool
+	cancel     func()
+	conf       drivers.Config
+	docker     dockerClient // retries on *docker.Client, restricts ad hoc *docker.Client usage / retries
+	hostname   string
+	auths      map[string]driverAuthConfig
+	pool       DockerPool
+	imageCache *Cache
 	// protects networks map
 	networksLock sync.Mutex
 	networks     map[string]uint64
 }
 
+// NewImageCleaner builds an evicter that loops every ImageCacheCleanInterval, checkes to see
+// if there is more space consumed then the maxSize allowed by configuration. If there is
+// NewImageCleaner checks the image cache for the list of evicitable images and then tries
+// in order of most evicitable to remove the image. If the disk consumption constraint is
+// satisfied the for loop breaks.
+func NewImageCleaner(context context.Context, dockerDriver *DockerDriver, maxSize int64) error {
+	opts := docker.RemoveImageOptions{}
+	duopts := docker.DiskUsageOptions{}
+	opts.Force = true
```

Comment:
Given that we track in-use, do we still need Force=true?

Reply:
force may cause issues, particularly if for whatever reason an image is a dependency of another function's image - should remove this.